### PR TITLE
ToDatacube from a collection now doesn't hide duplicit keys

### PR DESCRIPTION
**Pull request by KaliCZ**
Made Datacube constructor from a collection behave just like a dictionary does - throw exception for duplicit keys.

### CODE REVIEW

File: src/FuncSharp/Extensions/IEnumerableExtensions.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿using System;
+﻿
+using System;
```
Comment (marektresnak) [inline src/FuncSharp/Extensions/IEnumerableExtensions.cs]:
```suggestion
using System;
```

**Review by marektresnak — APPROVED** [review]
This looks like a safer choice to me, but it is a breaking, so that must be reflected when releasing a new version of FuncSharp.

**Comment by siroky** [comment]
@KaliCZ  Would you please add a test for this here https://github.com/siroky/FuncSharp/blob/master/src/FuncSharp.Tests/Extensions/IEnumerableExtensionsTests.cs?

**Comment by KaliCZ** [comment]
@siroky Got a test :)

**Comment by siroky** [comment]
@KaliCZ it's now released in version `4.3.0`.

**Pull request merged**